validator/src/main/java/com/amazon/aoc/validators/CWLogValidator.java

```diff
@@ -0,0 +1,114 @@
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.fge.jsonschema.main.JsonSchema;
+import com.github.fge.jsonschema.main.JsonSchemaFactory;
+import com.github.fge.jsonschema.report.ListReportProvider;
+import com.github.fge.jsonschema.report.LogLevel;
+import com.github.fge.jsonschema.report.ProcessingReport;
+import com.github.fge.jsonschema.util.JsonLoader;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.*;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+public class CWLogValidator implements IValidator {
+
+  protected String logStreamName = "otlp-logs";
+
+  private static final String LOGGROUPPATH = "/aws/ecs/otlp/%s/logs";
+
+  protected CloudWatchService cloudWatchService;
+  private static final int CHECK_INTERVAL_IN_MILLI = 30 * 1000;
+  private static final int CHECK_DURATION_IN_SECONDS = 2 * 60;
+  private static final int MAX_RETRY_COUNT = 12;
+  private static final int QUERY_LIMIT = 100;
+  private JsonSchema schema;
+  protected String logGroupName;
+
+  private Context context;
+
+  protected final ObjectMapper mapper = new ObjectMapper();
+
+  @Override
+  public void init(
+      Context context,
+      ValidationConfig validationConfig,
+      ICaller caller,
+      FileConfig expectedDataTemplate)
+      throws Exception {
+    this.context = context;
+    cloudWatchService = new CloudWatchService(context.getRegion());
+    logGroupName = String.format(LOGGROUPPATH, context.getTestingId());
+    MustacheHelper mustacheHelper = new MustacheHelper();
+    String templateInput = mustacheHelper.render(expectedDataTemplate, context);
+    JsonNode jsonNode = JsonLoader.fromString(templateInput);
+    JsonSchemaFactory jsonSchemaFactory =
+        JsonSchemaFactory.newBuilder()
+            .setReportProvider(new ListReportProvider(LogLevel.INFO, LogLevel.FATAL))
+            .freeze();
+    JsonSchema jsonSchema = jsonSchemaFactory.getJsonSchema(jsonNode);
+    this.schema = jsonSchema;
+    caller.callSampleApp();
+  }
+
+  @Override
+  public void validate() throws Exception {
+    RetryHelper.retry(
+        getMaxRetryCount(),
+        CHECK_INTERVAL_IN_MILLI,
+        true,
+        () -> {
+          Instant startTime =
+              Instant.now().minusSeconds(CHECK_DURATION_IN_SECONDS).truncatedTo(ChronoUnit.MINUTES);
+          fetchAndValidateLogs(startTime);
+        });
+  }
+
+  protected void fetchAndValidateLogs(Instant startTime) throws Exception {
+    List<OutputLogEvent> logEvents =
+        cloudWatchService.getLogs(
+            logGroupName, logStreamName, startTime.toEpochMilli(), QUERY_LIMIT);
+    if (logEvents.isEmpty()) {
+      throw new BaseException(
+          ExceptionCode.LOG_FORMAT_NOT_MATCHED,
+          String.format(
+              "[StructuredLogValidator] no logs found under log stream %s" + " in log group %s",
+              logStreamName, logGroupName));
```

Comment:
Don't use `String` class for logging. Use methods such a `log.info()` or `log.error()`.

Reply:
Just to clarify, you are saying to use log.error() in place of throwing exception?

Reply:
Ahhh, ignore this comment. I missed that this was being thrown in an exception. my aplogies.